py/client/pydeephaven/table.py

```diff
@@ -756,6 +756,45 @@ def where_not_in(self, filter_table: Table, cols: Union[str, List[str]]) -> Tabl
             DHError
         """
         return super(Table, self).where_not_in(filter_table, cols)
+    
+    def slice(self, start: int, stop: int) -> Table:
+        """Extracts a subset of a table by row positions into a new Table.
+
+        If both the start and the stop are positive, then both are counted from the beginning of the table.
+        The start is inclusive, and the stop is exclusive. slice(0, N) is equivalent to :meth:`~Table.head` (N)
+        The start must be less than or equal to the stop.
+
+        If the start is positive and the stop is negative, then the start is counted from the beginning of the
+        table, inclusively. The stop is counted from the end of the table. For example, slice(1, -1) includes all
+        rows but the first and last. If the stop is before the start, the result is an empty table.
+
+        If the start is negative, and the stop is zero, then the start is counted from the end of the table,
+        and the end of the slice is the size of the table. slice(-N, 0) is equivalent to :meth:`~Table.tail` (N).
+
+        If the start is negative and the stop is negative, they are both counted from the end of the
+        table. For example, slice(-2, -1) returns the second to last row of the table.
```

Comment:
For the docs to render well, do the `slice` method calls need backticks or other formatting?

Reply:
Note that this is the exact same documentation string as is in the server side - I'm sure it could do with some improvement, but I'm not sure how much rewrite you want on both to land this feature?